### DISCUSSION TITLE
Confusion about Immediate

### DISCUSSION THREAD

**Discussion by percybolmer** [rabbitmq/amqp091-go#175] [Q&A] [answered]
Hey, I am writing a blog post about RabbitMQ in Go, and once doing this I really try to dive deep into understanding everything.

I am a bit confused about the `immediate` parameter in the Publish functions, that accepts an Immediate flag.
Is this left for backward compatibility and can be ignored if using RabbitMQ v3+? 

I've noticed that immediate is removed after that.
https://blog.rabbitmq.com/posts/2012/11/breaking-things-with-rabbitmq-3-0

**Comment by lukebakken** [comment, marked as answer]
> Is this left for backward compatibility and can be ignored if using RabbitMQ v3+?

Yes. It is there because this library can be used with other AMQP 0.9.1 servers that may support the flag.

**Reply by percybolmer** [reply]
That's what I suspected! Thanks for clarifying
